docker.gradle

```diff
@@ -92,34 +91,35 @@ task copyContainerFiles(type: Copy) {
 task buildImage(type: DockerBuildImage) {
   dependsOn copyContainerFiles
   inputDir = file(build_context)
-  tag = System.getenv('DOCKER_REPO_URL') + '/uaa:' + version
+  images.add(System.getenv('DOCKER_REPO_URL') + '/uaa:' + version)
   buildArgs = [MAINTAINER_EMAIL: System.getenv('MAINTAINER_EMAIL'), VERSION: version]
 }
 
 task createUaaContainer(type: DockerCreateContainer) {
   dependsOn buildImage
-  targetImageId { buildImage.getImageId() }
+  targetImageId buildImage.imageId
   containerName = System.getenv('DOCKER_CONTAINER_NAME')
-  portBindings = ['8080:8080']
-  env = ["CATALINA_OPTS='-Dspring.profiles.active=default'", 
-         "ENFORCE_CLIENT_ASSERTION_HEADER=" + System.getenv('ENFORCE_CLIENT_ASSERTION_HEADER'),
-         "KEY_PROVIDER_SERVICE_URL=" + System.getenv('KEY_PROVIDER_SERVICE_URL'),
-         "KEY_PROVIDER_TOKEN_URL=" + System.getenv('KEY_PROVIDER_TOKEN_URL'),
-         "KEY_PROVIDER_CLIENT_ID=" + System.getenv('KEY_PROVIDER_CLIENT_ID'),
-         "KEY_PROVIDER_CLIENT_SECRET=" + System.getenv('KEY_PROVIDER_CLIENT_SECRET'),
-         "KEY_PROVIDER_DEFAULT_INSTANCE=" + System.getenv('KEY_PROVIDER_DEFAULT_INSTANCE')]
+  hostConfig.portBindings = ['8080:8080']
+  envVars.putAll([
+          'CATALINA_OPTS': '-Dspring.profiles.active=default',
+          'ENFORCE_CLIENT_ASSERTION_HEADER': System.getenv('ENFORCE_CLIENT_ASSERTION_HEADER'),
```

Comment:
These changes, line 105 through 110, should be made on predix_prunes/jwt_grant_75.6.0 branch, not on rc_75.6.0 directly.

Overall, dockerization changes to dependencies.gradle, docker.gradle, uaa-docker.yml should be merged first to predix_prunes/jwt_grant_75.6.0.  Then both predix_prunes/jwt_grant_75.6.0 and predix_extensions_75.6.0 should be merged to rc_75.6.0

Reply:
That makes sense. Will do.

Reply:
Actually, it is messy. We seem to have done the dockerization work in predix_extensions, then somebody copied it over to the jwt_grant branch, but then all updates to dockerization have been done only for the predix_extensions branch. So jwt branch has a docker.gradle that is quite out of date with predix_extensions. And there is not 'merging' the changes from predix_extensions into jwt_grant - so we will have to copy the changes, which is wrong. So since the dockerization has to happen in a way that works across branches (predix_extensions, jwt_grant, maybe others), it might make sense to have another branch for dockerization. I guess for now, I will copy the fixes over to the jwt branch from the predix_extensions branch. So that we can build a docker container from the jwt_grant branch also even though we don't do that now.

Reply:
I agree it's messy... we will not be able to build docker image from /predix_prunes/jwt_grant_75.6.0 as it doesn't contain necessary Dockerfile and Jenkinsfile files...  However, changes to docker.gradle and uaa-docker.yml are specific to the device authentication and should be done on the /predix_prunes/jwt_grant_xxx branch to support future merges.